library/common/templates/_service.tpl

```diff
@@ -19,5 +18,19 @@ of the main Service and any additionalServices.
         {{- include "common.classes.service" $ -}}
       {{- end }}
     {{- end }}
+    {{- /* Generate TrueNAS SCALE app services as required v1 */ -}}
+    {{- if and .Values.appAdditionalServicesEnabled .Values.appAdditionalServices -}}
+      {{- range $name, $srv := .Values.appAdditionalServices }}
+        {{- if $srv.enabled -}}
+          {{- print ("---") | nindent 0 -}}
+          {{- $serviceValues := $srv -}}
+          {{- if not $serviceValues.nameSuffix -}}
+            {{- $_ := set $serviceValues "nameSuffix" $name -}}
+          {{ end -}}
+          {{- $_ := set $ "ObjectValues" (dict "service" $serviceValues) -}}
+          {{- include "common.classes.service" $ -}}
+        {{- end }}
+      {{- end }}
+    {{- end }}
```

Comment:
can we move this section to a seperate file under /custom?

Reply:
done

Reply:
awesome!